src/main/java/dk/kb/storage/storage/DsStorage.java

```diff
@@ -841,7 +844,7 @@ public MappingDto getMappingByReferenceId(String referenceId) throws Exception {
  
            try (ResultSet rs = stmt.executeQuery();) {
                if (!rs.next()) {
-                 return null; //ID not found
+                 throw new InvalidArgumentServiceException("No mapping found for referenceId:"+referenceId);
```

Comment:
Is it too offensive to throw an exception here. Couldn't we be in a situation where we need the mapping, but the table haven't been updated yet?

Reply:
I think it is the exception this situation will happen. I only can see it happen from fullpost view on a record that does not have a kalturaId. Ie. the post will not be in index and the user should be able to be on the fullpost page.